projects/website/markdown/docs/plugins/creating-plugins.mdx

```diff
@@ -56,6 +56,57 @@ In order for Tuist to locate the templates for a plugin, they must be placed in
 └── ...
 ```
 
+## ResourceSynthesizers
+
+ResourceSynthesizer plugins are for sharing & reusing templates for synthesizing resources. If you want to use one of the predefined resource synthesizers, the template must also adhere to a specific naming.
+
+For example if you initialize `ResourceSynthesizer` with `.strings(plugin: "MyPlugin")` then the template must be called `Strings.stencil`.
+
+There are more types, so the naming is:
+
+- `strings` => `Strings.stencil`
+- `assets` => `Assets.stencil`
+- `plists` => `Plists.stencil`
+- `fonts` => `Fonts.stencil`
+- `coreData` => `CoreData.stencil`
+- `interfaceBuilder` => `InterfaceBuilder.stencil`
+- `json` => `JSON.stencil`
+- `yaml` => `YAML.stencil`
+
+You can also create a `ResourceSynthesizer` with `.custom`. In this case the template should be of the same name as `resourceName`.
+
+In order for Tuist to locate the templates for a plugin, they must be placed in the same directory as the `Plugin.swift` manifest and in a directory named `ResourceTemplates`.
+
+```
+.
+├── ...
+├── Plugin.swift
+├── ResourceTemplates
```

Comment:
Could we go a few levels deeper here and show what a full fledged strings example might look like in terms of layout / file names?

Reply:
good idea, I'll add it